src/lib/Scenes/Artwork/Artwork.tsx

```diff
@@ -340,24 +336,71 @@ export const ArtworkRenderer: React.SFC<{ artworkID: string; safeAreaInsets: Saf
     <RetryErrorBoundary
       render={({ isRetry }) => {
         return (
-          <QueryRenderer<ArtworkQuery>
-            environment={defaultEnvironment}
-            query={graphql`
-              query ArtworkQuery($artworkID: String!) {
-                artwork(id: $artworkID) {
-                  ...Artwork_artwork
-                }
-              }
-            `}
-            variables={{ artworkID }}
-            cacheConfig={{
-              // Bypass Relay cache on retries.
-              ...(isRetry && { force: true }),
-            }}
-            render={renderWithLoadProgress(ArtworkContainer, others)}
-          />
+          <Theme>
+            <ProvideScreenDimensions>
+              <QueryRenderer<ArtworkQuery>
+                environment={defaultEnvironment}
+                query={graphql`
+                  query ArtworkQuery($artworkID: String!) {
+                    artwork(id: $artworkID) {
+                      ...Artwork_artwork
+                    }
+                  }
+                `}
+                variables={{ artworkID }}
+                cacheConfig={{
+                  // Bypass Relay cache on retries.
+                  ...(isRetry && { force: true }),
+                }}
+                render={renderWithPlaceholder({
+                  Container: ArtworkContainer,
+                  initialProps: others,
+                  renderPlaceholder: () => <ArtworkPlaceholder />,
+                })}
+              />
+            </ProvideScreenDimensions>
+          </Theme>
         )
       }}
     />
   )
 }
+
+const ArtworkPlaceholder: React.FC<{}> = ({}) => {
+  if (process?.env?.NODE_ENV === "test") {
```

Comment:
I think I get the purpose of this, but let me make sure I understand. Normally we would mock the module for this component in our test file, but we would prefer to keep this loading view in the same file, making it impossible to mock with Jest. Did I get that right? Unusual but justified, we should add a comment 👍

Reply:
I removed this and made the RaggedText placeholder deterministic to avoid confusion and prevent snapshot churn.